### PR TITLE
Drop unsupported executables when installing

### PR DESCRIPTION
**Pull request by sl-mlx**
When installing a BFB to the boot partitions, image versions
that are not supported by the current platform may be installed,
even though they'll never actually execute. In some cases, this may
cause the BFB to fail installation due to size limitations.

So, when installing, drop any images for hardware later than the current
platform.

### CODE REVIEW

**Review by lsun100 — APPROVED** [review]

File: mlxbf-bootctl.c [hunk 1]
```diff
@@ -112,8 +112,11 @@ void die(const char* fmt, ...)
 #define LIFECYCLE_STATE_PATH "lifecycle_state"
 #define SECURE_BOOT_FUSE_STATE_PATH "secure_boot_fuse_state"
 
+#define INPUT_BFB_MAX_SIZE (20 * 1024 * 1024)
 #define MMC_BOOT_PARTITION_MAX_SIZE (4 * 1024 * 1024)
 
+#define DMI_PROCESSOR_INFO_TYPE 4
+
```
Comment (lsun100) [inline mlxbf-bootctl.c:119]: These two lines seems not used?

File: mlxbf-bootctl.c [hunk 2]
```diff
@@ -539,20 +651,15 @@ void write_bootstream(const char *bootstream, const char *bootfile, int flags)
...
   // Copy the bootstream to the bootfile device
-  int ifd = open(bootstream, O_RDONLY);
-  if (ifd < 0)
-    die("%s: %m", bootstream);
+  void *ibuf = malloc(MMC_BOOT_PARTITION_MAX_SIZE);
+  void *inidx = ibuf;
+  uint8_t padbuf[8] = {0}; // There are at most 8 pad bytes.
+  if (ibuf == NULL)
+    die("out of memory");
+  size_t bytes_left = read_bootstream_to_buffer(bootstream, ibuf, MMC_BOOT_PARTITION_MAX_SIZE, 1);
```
Comment (lsun100) [inline mlxbf-bootctl.c:659]: why value 1 here, is it still working in progress?  Looks like version bitmap hasn;t been updated yet.

File: mlxbf-bootctl.c (hunk 1 above)
Comment (sl-mlx) [reply mlxbf-bootctl.c:119]: Forgot to remove it for this patch. This is WIP.

File: mlxbf-bootctl.c (hunk 2 above)
Comment (sl-mlx) [reply mlxbf-bootctl.c:659]: Yes, it's still WIP. I'm adding a function to figure out the correct version to pass into that field.

File: mlxbf-bootctl.c [hunk 3]
```diff
@@ -60,6 +69,27 @@ void die(const char* fmt, ...)
...
+    unsigned long major:4;
+    unsigned long minor:4;
+    unsigned long reserved:4;
+    unsigned long next_img_ver:4;
+    unsigned long cur_img_ver:4;
+    unsigned long hdr_len:4;
+    unsigned long image_id:8;
+    unsigned long image_len:32;
+    unsigned long image_crc:32;
+    unsigned long following_images:64;
```
Comment (lsun100) [inline mlxbf-bootctl.c:90]: Need to update this field when trimming images.

Comment (sl-mlx) [reply mlxbf-bootctl.c:90]: Done. Updated `next_img_ver` as well.

**Pull request merged**